bin/update_dependencies.py

```diff
@@ -12,21 +13,39 @@
 
 # CUSTOM_COMPILE_COMMAND is a pip-compile option that tells users how to
 # regenerate the constraints files
-os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_constraints.py"
+os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_dependencies.py"
 subprocess.check_call([
```

Comment:
I just realised... this command is implicitly run in python3.8, because that's what my venv is in on my mac >.< (maybe a little sloppy, sorry!). Could it be made explicit? Perhaps `38` and `39` should be added to the list of versions and the default constraints.txt uses the final entry of this list? Then it should also be generated inside the docker container, so we're sure which version of Python is running it.

Reply:
Shall we use a default `constraints.txt` at all ?

Reply:
That's a great point. Maybe we don't need it at all, in fact!

Reply:
Oh, well actually our code does assert that it exists, currently

Reply:
It's also maybe useful as a starting point if somebody wanted to customise their constraints. But just copying the latest Python we support should do

Reply:
Done. I had to modify a test so I updated all dependencies. Anyway, I think I'll have to rebase on #489 anyway.